src/main/java/net/sf/mzmine/modules/peaklistmethods/io/csvexport/CSVExportTask.java

```diff
@@ -75,35 +78,53 @@ public double getFinishedPercentage() {
     }
 
     public String getTaskDescription() {
-        return "Exporting peak list " + peakList + " to " + fileName;
+        return "Exporting peak list(s) " 
+                + Arrays.toString(peakLists) + " to CSV file(s)";
     }
 
     public void run() {
 
         setStatus(TaskStatus.PROCESSING);
 
-        // Open file
-        FileWriter writer;
-        try {
-            writer = new FileWriter(fileName);
-        } catch (Exception e) {
-            setStatus(TaskStatus.ERROR);
-            setErrorMessage("Could not open file " + fileName + " for writing.");
-            return;
+        // Shall export several files?
+        boolean substitute = fileName.getPath().contains(plNamePattern);
+
+        // Total number of rows
+        for (PeakList peakList: peakLists) {
+            totalRows += peakList.getNumberOfRows();
         }
 
-        // Get number of rows
-        totalRows = peakList.getNumberOfRows();
+        // Process peak lists
+        for (PeakList peakList: peakLists) {
 
-        exportPeakList(peakList, writer);
+            // Filename
+            File curFile = fileName;
+            if (substitute) {
+                curFile = new File(fileName.getPath().replaceAll(
+                        Pattern.quote(plNamePattern), peakList.getName()));
+            }
 
-        // Close file
-        try {
-            writer.close();
-        } catch (Exception e) {
-            setStatus(TaskStatus.ERROR);
-            setErrorMessage("Could not close file " + fileName);
-            return;
+            // Open file
+            FileWriter writer;
+            try {
+                writer = new FileWriter(curFile);
+            } catch (Exception e) {
+                setStatus(TaskStatus.ERROR);
+                setErrorMessage("Could not open file " + curFile 
+                        + " for writing.");
+                return;
+            }
+
+            exportPeakList(peakList, writer, curFile);
+

```

Comment:
After exportPeakList() we should add a check if the task has been canceled
```
if (isCanceled()) return;
```
Otherwise the task will just continue with next peak list